src/cert.rs

```diff
@@ -16,24 +16,49 @@ use crate::der::Tag;
 use crate::x509::{remember_extension, set_extension_once, Extension};
 use crate::{der, signed_data, Error};
 
+/// An enumeration indicating whether a [`Cert`] is a leaf end-entity cert, or a linked
+/// list node from the CA `Cert` to a child `Cert` it issued.
 pub enum EndEntityOrCa<'a> {
+    /// The [`Cert`] is a leaf end-entity certificate.
     EndEntity,
+    /// The [`Cert`] is an issuer certificate, and issued the referenced child `Cert`.
     Ca(&'a Cert<'a>),
 }
 
+/// A parsed X509 certificate.
```

Comment:
Is the intent here that a `Cert` can only be created in webpki and passed to caller code as a parameter? If so, perhaps we should make this `#[non_exhaustive]` to avoid allowing downstream code to create their own `Cert` values?

Have we previously exposed `untrusted::Input<'_>` types in the public interface, and do we think that is the interface we want to expose to downstream callers?

Reply:
I waffled on both points a bit in some earlier [self-review](https://github.com/rustls/webpki/pull/66#discussion_r1196836037).

Looking through the existing exported types there's no `untrusted::Input` so I think it's clear we should avoid that.

Practically speaking that makes exporting the `Cert` type unhelpful since it's a big bag of `untrusted::Input`. With that in mind I think it's better to try to identify the most relevant fields and pass only those through to the CrlProvider. 

I'll rework :+1:

Reply:
Could also still be the `Cert` type, but with methods instead of public fields?

Reply:
> Could also still be the Cert type, but with methods instead of public fields? I don't see much point in inventing a wrapper type to expose to other code if we just need a higher-level abstraction here.

Reworked to handle as suggested.

Reply:
> I'll rework that separately from this branch sweat

https://github.com/rustls/webpki/pull/83

I will resolve conflicts as these branches land.